tests/test_downloader.py

```diff
@@ -0,0 +1,293 @@
+#!/usr/bin/env python3
+#
+# Copyright 2021 Red Hat, Inc.
+# All Rights Reserved.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License"); you may
+#    not use this file except in compliance with the License. You may obtain
+#    a copy of the License at
+#
+#         http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+#    WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+#    License for the specific language governing permissions and limitations
+#    under the License.
+#
+import json
+import os
+import shutil
+from argparse import Namespace
+from dataclasses import dataclass
+from unittest import TestCase
+from unittest.mock import patch
+
+from requests.exceptions import RequestException
+
+from znoyder.downloader import (CONTENT_ENDPOINT, GITHUB_API_URL,
+                                OPENDEV_API_URL, REPO_ENDPOINT, download_file,
+                                download_files_parallel, download_zuul_config,
+                                get_raw_url_files_in_repository, main)
+
+
+@dataclass
+class MockHTMLResponse:
+    """Mock an HTML response from requests.get."""
+    status_code: int
+    text: str
+    url: str
+    content: bytes
+
+
+class TestDownloader(TestCase):
+    """Test the downloader module."""
+
+    def setUp(self):
+        self.dest_dir = "test_downloader_dest_dir"
+        os.makedirs(self.dest_dir)
```

Comment:
Maybe using https://docs.python.org/3/library/tempfile.html#tempfile.TemporaryDirectory would be a better option here?

Alternatively, I would even consider creating in memory directory with something like https://docs.pyfilesystem.org/en/latest/reference/memoryfs.html, but that may be an overkill :-)

Reply:
Yes, using tempfile seems a good idea here, I'll change it

Reply:
I've updated the PR using a TemporaryDirectory